src/utils/extractors.ts

```diff
@@ -0,0 +1,69 @@
+import { isArray } from 'lodash'
+import { IEndpointMiddleware, ISwaggerMiddlewareConfig } from '../types/interfaces'
+
+function extractMiddlewareName(configMiddleware: ISwaggerMiddlewareConfig): string {
+	const tempName = configMiddleware.middlewareName.replace('<', '').replace('>', '')
+
+	return `${tempName.charAt(0).toUpperCase() + tempName.slice(1).toLowerCase()}`
+}
+
+/**
+ * Works only with first middleware argument and prints his value if it consists of a primitive value array or an object containing such arrays
+ * @param endpointMiddleware object containing all necessary information about actual middleware
+ * or undefined in case the middleware is not used in a given endpoint.
+ * @param configMiddleware config information of a middleware wich should be processed
+ * @return { string } middleware's description
+ * */
+export const firstVersionExtractor = (endpointMiddleware: IEndpointMiddleware | undefined, configMiddleware: ISwaggerMiddlewareConfig): string => {
+	if (!endpointMiddleware || endpointMiddleware.middlewareArguments.length === 0) {
+		return `NO`
+	}
+
+	const permissionsResult = `${extractMiddlewareName(configMiddleware)}: `
+
+	const { value } = endpointMiddleware.middlewareArguments[0]
+
+	if (isArray(value)) {
+		return `${permissionsResult} ${value}`
+	}
+
+	return `${permissionsResult}<ul>${Object.keys(value)
+		.map((key) => {
+			let result = ''
+
+			Object.values(value[key]).forEach((array) => {
+				if (isArray(array)) {
+					result += `<li>${key}${array.length > 0 ? `: [${array.join(', ')}]` : ''}</li>`
+				}
+			})
+
+			return result
+		})
+		.join('')}</ul>`
+}
+
+/**
+ * Default extractor
+ * @param endpointMiddleware object containing all necessary information about actual middleware
+ * or undefined in case the middleware is not used in a given endpoint.
+ * @param configMiddleware config information of a middleware wich should be processed
+ * @return { string } middleware's description
+ * */
+export const defaultExtractor = (endpointMiddleware: IEndpointMiddleware | undefined, configMiddleware: ISwaggerMiddlewareConfig) => {
+	let value = 'false'
+	const middlewareName = extractMiddlewareName(configMiddleware)
```

Comment:
@kot-1999 middlewareName  should contain endpointMiddleware param. And if endpointMiddleware can be undefined it should be at second param after configMiddleware